### PR TITLE
Improvements for sudo for deb systems

### PR DESCRIPTION
**Pull request by juliogonzalez**
# Description

- Ubuntu: document how to onboard with the bundle as a regular user 
- Debian: document how to onboard with the bundle or regular salt as  a regular user

# Target branches

Which documentation version does this PR apply to?

- [x] Master (Default)
- [ ] Manager-4.2 <--- I think you need to backport it, even if the bundle is optional? Check with @vzhestkov 
- [ ] Manager-4.1
- [ ] Manager-4.0

# Links

No reported issue. Detected while testing Ubuntu 22.04

### CODE REVIEW

File: modules/client-configuration/pages/clients-ubuntu-old.adoc [hunk 1]
```diff
@@ -228,7 +229,7 @@ Grant [command]``sudo`` access to the user by adding this line at the end of the
 Replace [systemitem]``<user>`` with the name of the user that is bootstrapping the client in the {webui}:
 +
 ----
-<user>  ALL=NOPASSWD: /usr/bin/python, /usr/bin/python2, /usr/bin/python3
+<user>  ALL=NOPASSWD: /usr/bin/python, /usr/bin/python2, /usr/bin/python3, /var/tmp/venv-salt-minion/bin/python
```
Comment (vzhestkov) [inline modules/client-configuration/pages/clients-ubuntu-old.adoc:232]: Same here

File: modules/client-configuration/pages/clients-debian.adoc [hunk 2]
```diff
@@ -159,6 +159,34 @@ include::snippets/trust_gpg.adoc[]
...
++
+----
+sudo visudo
+----
++
+Grant [command]``sudo`` access to the user by adding this line at the end of the [filename]``sudoers`` file.
+Replace [systemitem]``<user>`` with the name of the user that is bootstrapping the client in the {webui}:
++
+----
+<user>  ALL=NOPASSWD: /usr/bin/python, /usr/bin/python2, /usr/bin/python3, /var/tmp/venv-salt-minion/bin/python
```
[the remark this answers is not in the document]
Comment (juliogonzalez) [reply modules/client-configuration/pages/clients-debian.adoc:180]: @vzhestkov: what if someone disabled the onboarding with the bundle (possible), and disabled adding the bundle to the bootstrap repo?

File: .changelog [hunk 3]
```diff
@@ -4,6 +4,9 @@
 # - Fixed error in Bat section of Upgrade Guide (bsc#1234567)
 # For guidelines: https://en.opensuse.org/openSUSE:Creating_a_changes_file_(RPM)#Changelog_section_.28.25changelog.29
 
+- Ubuntu: document how to onboard with the bundle as a regular user
+- Debian: document how to onboard with the bundle or regular salt as
+  a regular user
```
[the remark this answers is not in the document]
Comment (juliogonzalez) [reply .changelog:9]: > * Wondering whether we actually must mention Salt?

Why not? This applies only to salt.

> * Shall we add "in the Client Configuration guide"?

In the end we have way too many entries with "in the ...", "in the...", adding lenthg to all lines.

FMPOV, if you think this is needed, then whoever packages, should do a clean and group stuff:

```
- Configuration guide:
  * Change A
  * Change B
 - Administration Guide:
   * Change C
   * CHange D
```

**Review by 0rnela — APPROVED** [review]

File: modules/client-configuration/pages/clients-debian.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (juliogonzalez) [reply modules/client-configuration/pages/clients-debian.adoc:180]: Not really. If you do it you break Debian9 (still not EoL from a SUMA POV, and I am sure people will still use it for a while, as it has LTS even if we don't support it). This doc is common for all of them.

And for Ubuntu 18.04 is still using `python2` IIRC (not to menthion 16.04, but that one is already EoL).

Or you need to tell people to choose whatever is correct.

So I suggest leaving it this way, except for systems where we *only* have the bundle.

File: .changelog (hunk 3 above)
[the remark this answers is not in the document]
Comment (keichwa) [reply .changelog]: Sure, and I'd even ask for less (and more)! ;)

Detailed documentation are best tracked in an appendix of the book.  In the RPM changes I'd just list general changes (packaging issues and output improvements caused by tool changes) plus a list of numbers of fixed bugs.  Novell did this very well in the early years of 21st cent., but people usually think that that's too much work...

**Comment by keichwa** [comment]
@juliogonzalez and @vzhestkov --I do not understand all your discussions, but I think this PR is now ready to be merged (and then backported to 4.2). If this is not the case please veto!

**Review by jcayouette — APPROVED** [review]

**Comment by 0rnela** [comment]
@vzhestkov @juliogonzalez @jcayouette @keichwa -- what is the status of this PR?

**Review by vzhestkov — APPROVED** [review]
Julio's explanations are fine for me. :+1:

**Comment by keichwa** [comment]
By now, we only have check marks.  Thus I assume it is ready to be merged (and then backported to 4.3).  I'm not sure whether 4.2 is still wanted or needed.

**Comment by vzhestkov** [comment]
I think, it's better to backport it to `4.2` also, as the users can enable the bundle for 4.2.

**Comment by keichwa** [comment]
> I think, it's better to backport it to `4.2` also, as the users can enable the bundle for 4.2.

Yes, I agree!

**Pull request merged**